src/tools/compiletest/src/runtest.rs

```diff
@@ -203,7 +202,7 @@ pub fn compute_stamp_hash(config: &Config) -> String {
         None => {}
     }
 
-    if let Ui = config.mode {
+    if config.mode == TestMode::Ui {
         config.force_pass_mode.hash(&mut hash);
```

Comment:
Remark: as a concrete example, `tests/ui/` is both `TestMode::Ui` and `TestSuite::Ui`, but these are very different things and must not be mixed up, because a test mode can correspond to multiple test suites.

(`TestSuite` is something I want to introduce over stringly-typed test suite names in a follow-up.)

Reply:
https://github.com/rust-lang/rust/pull/143870